src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String BUY = "buy";
+    private static final String SUPPLY = "supply";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String receivingFile = readFromFile(fromFileName);
+        String report = calculateReport(receivingFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                builder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read data from the file: " + fromFileName, e);
+        }
+        return builder.toString();
+    }
+
+    private String calculateReport(String fileContents) {
+        StringBuilder builder1 = new StringBuilder();
+        String[] data = fileContents.split(System.lineSeparator());
+        int supply = 0;
+        int buy = 0;
+        for (String element : data) {
+            String[] elements = element.split(",");
```

Comment:
```suggestion
            String[] elements = element.split(COMMA);
```
now you can use COMMA constant